db/src/main/kotlin/edu/artic/db/MoshiAdapters.kt

```diff
@@ -63,4 +65,36 @@ class ZonedDateTimeAdapter {
 
     @FromJson
     fun fromText(text: String): ZonedDateTime = ZonedDateTime.parse(text, formatter)
-}
\ No newline at end of file
+}
+
+/**
+ * Parses floor as an integer. We default to [Int.MIN_VALUE] as 0 is a valid floor.
+ */
+class FloorAdapter {
+
+    @ToJson
+    fun toText(@Floor floor: Int): String = floor.toString()
+
+    @FromJson
+    @Floor
+    fun fromText(text: String?): Int {
+        return if (text != null) {
+            try {
+                return text.toInt()
+            } catch (e: NumberFormatException) {
+                if (text == "LL") {
```

Comment:
Explicit comment above this line: `"LL" stands for 'Lower level'` or something like that.

Reply:
Updated.